x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/render_table_columns/render_endpoint/render_endpoint.tsx

```diff
@@ -0,0 +1,226 @@
+/*
```

Comment:
This is still not ideal, the render functions instead of components is not great but what was asked for was a generic component that takes the endpoint an abstracts the differences. We still have specific rendering for each type of endpoint just in functions instead of components now.

I'll let @sphilipse advise as well but I think this look something more like:

```typescript
interface EndpointInfoProps {
  endpoint: InferenceAPIConfigResponse;
}
const EndpointInfo = ({ endpoint }: EndpointInfoProps) => {
  return (
    <EuiFlexGroup gutterSize="s" direction="column">
      <EuiFlexItem grow={false}>
        <strong>{endpoint.model_id}</strong>
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
         <EndpointModelInfo endpoint={endpoint} />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}

const EndpointModelInfo = ({ endpoint }: EndpointInfoProps) => {
// Endpoints with Badges
// else
// endpoints with just Text
// etc. 
}
```

Reply:
Yes, correct. We're duplicating a lot of rendering code because we're switching on full components, instead of trying to abstract the difference. We should also be defaulting to using React Components instead of plain functions when returning HTML.

Reply:
I changed this file to remove duplicates. Let me know whether this change makes sense now. With the current changes, we are now returning strings instead of HTML for different service providers. As a result, I kept plain functions for those providers. The rest of the changes are implemented as Components.